DIIS in HF + Fixed FCI bugs (correlation_ratio)

scemama opened the pull request.

TApplencourt commented on scripts/ezfio_interface/qp_convert_output_to_ezfio.py:
```diff
@@ -309,8 +309,16 @@ def parse_str(pseudo_str):
-
+            x = [[coef_n_zeta.split() for coef_n_zeta in l.split('\n')] \
+               for l in array_party[1:] ]
+            x = []
```
Juste add `filter(None,coeff_n_zera.split())` and remove the line (314-320)

scemama replied:
Changed:
            x = [[ filter(None,coef_n_zeta.split()) for coef_n_zeta in l.split('\n')] for l in array_party[1:] ]

TApplencourt commented on scripts/ezfio_interface/qp_convert_output_to_ezfio.py:
```diff
@@ -421,3 +444,4 @@ def get_full_path(file_path):
         print file_, 'recognized as', str(res_file).split('.')[-1].split()[0]
 
     write_ezfio(res_file, ezfio_file)
+    os.system("qp_run save_ortho_mos "+ezfio_file)
```
So know to run this script we need to have a compiled version this binary.
Maybe make it more explicit

scemama replied:
Yes, but if we forget to do this step, all the QP after will be wrong.
I added a warning message in case it does not work.

TApplencourt commented on plugins/Hartree_Fock/SCF.irp.f:
```diff
@@ -5,6 +5,7 @@ program scf
 ! output: hartree_fock.energy
 ! optional: mo_basis.mo_coef
   END_DOC
+!  call debug
```
Maybe remove this line, or we can put a environement variable to control this kind one thing

scemama replied:
OK

TApplencourt commented on plugins/Hartree_Fock/SCF.irp.f:
```diff
@@ -34,21 +35,30 @@ subroutine create_guess
   endif
 end
 
+ao_to_mo
```
?

scemama replied:
Oops...

TApplencourt commented on plugins/Hartree_Fock_SlaterDressed/at_nucl.irp.f:
```diff
@@ -0,0 +1,66 @@
+!      expo = slater_expo(i)*slater_expo(i)*((x*x) + (y*y) + (z*z))
+!      if (expo > 160.d0) cycle
+!      expo = dsqrt(expo)
+      expo = slater_expo(i) * dsqrt((x*x) + (y*y) + (z*z))
```
Maybe remove the commented code ?

scemama replied:
Not finished yet. Will be cleaned in a few days

TApplencourt commented on plugins/Hartree_Fock_SlaterDressed/integrals.irp.f:
```diff
@@ -0,0 +1,624 @@
+  ss = k*ss
+
+! Print result
+  write(*,*) ss
```
Debug output or mandatory?

scemama replied:
Not finished yet. Will be cleaned in a few days

TApplencourt commented on plugins/Hartree_Fock_SlaterDressed/integrals.irp.f:
```diff
@@ -0,0 +1,624 @@
+end
+!*****************************************************************************
+
+!TODO
```
Do we really want to merge TODO thing in the main-branch?

scemama replied:
Yes. We need a quick fix for Anouar :-s

TApplencourt commented on plugins/MPI/utils.irp.f:
```diff
@@ -0,0 +1,68 @@
+  implicit none
+  PROVIDE MPI_Initialized
+  IRP_IF MPI
+      integer :: ierr
```
The Provide MPI_Init doesn't need to go in the IRP_IF to be cleaner?
Or beter create the function one if IRP_IF MPI (don't know if it's possible)

TApplencourt commented on plugins/MRPT/mrpt.irp.f:
```diff
@@ -1,38 +0,0 @@
-  print *, '           .~      +-             '
-  print *, '           . .... : .             '
-  print *, '            -------~              '
-  print *, ''
```
Maybe we can add some UTF8 character !

scemama replied:
This one has been removed from the repo

scemama commented:
Some problems when the installation fails. We had to remove everything and start again from scratch.
Development mode is better.

Le 16 juin 2017 19:22:40 GMT+02:00, Thomas Applencourt <notifications@github.com> a écrit :
>TApplencourt commented on this pull request.
>
>Amazing work. Just a few remarks
>
>> @@ -2,26 +2,18 @@
> # -*- coding: utf-8 -*-
> """configure
> 
>-Usage: configure <config_file> (--production | --development)
>
>Why ?
>
>> @@ -528,7 +517,7 @@ def create_ninja_and_rc(l_installed):
>     qp_create_ninja = os.path.join(QP_ROOT, "scripts", "compilation",
>                                    "qp_create_ninja.py")
> 
>-    l = [qp_create_ninja, "create"] + sys.argv[1:]
>+    l = [qp_create_ninja, "create", "--development"] + sys.argv[1:]
> 
>
>So we use only development mode now?
>
>> @@ -1,305 +0,0 @@
>-open Qputils;;
>
>Can we have a little documentation on how to use ddci ? And what is
>DDCI. 
>Nobody use ORCA =D
>
>- Do we need some MO?
>- What binary we need to run with what module ? 
>
>Maybe this ocam file need to be in a Module somehow...
>
>> @@ -309,8 +309,16 @@ def parse_str(pseudo_str):
>             array_l_max_block.append(l_max_block)
>             array_z_remove.append(z_remove)
> 
>-            matrix.append([[coef_n_zeta.split()[1:] for coef_n_zeta in
>l.split('\n')] for l in array_party[1:]])
>-
>+            x = [[coef_n_zeta.split() for coef_n_zeta in
>l.split('\n')] \
>+               for l in array_party[1:] ]
>+            x = []
>
>Juste add `filter(None,coeff_n_zera.split())` and remove the line
>(314-320)
>
>> @@ -421,3 +444,4 @@ def get_full_path(file_path):
>  print file_, 'recognized as', str(res_file).split('.')[-1].split()[0]
> 
>     write_ezfio(res_file, ezfio_file)
>+    os.system("qp_run save_ortho_mos "+ezfio_file)
>
>So know to run this script we need to have a compiled version this
>binary.
>Maybe make it more explicit
>
>> @@ -5,6 +5,7 @@ program scf
> ! output: hartree_fock.energy
> ! optional: mo_basis.mo_coef
>   END_DOC
>+!  call debug
>
>Maybe remove this line, or we can put a environement variable to
>control this kind one thing
>
>> @@ -34,21 +35,30 @@ subroutine create_guess
>   endif
> end
> 
>+ao_to_mo
>
>?
>
>> +  BEGIN_DOC
>+! Values of the Slater orbitals (1) at the nucleus (2)
>+  END_DOC
>+  integer :: i,j,k
>+  double precision :: x,y,z,expo,poly, r
>+
>+  do k=1,nucl_num
>+    do i=1,nucl_num
>+      x = nucl_coord(i,1) - nucl_coord(k,1)
>+      y = nucl_coord(i,2) - nucl_coord(k,2)
>+      z = nucl_coord(i,3) - nucl_coord(k,3)
>+
>+!      expo = slater_expo(i)*slater_expo(i)*((x*x) + (y*y) + (z*z))
>+!      if (expo > 160.d0) cycle
>+!      expo = dsqrt(expo)
>+      expo = slater_expo(i) * dsqrt((x*x) + (y*y) + (z*z))
>
>Maybe remove the commented code ?
>
>> +
>+  x = dsqrt(expSla**2/(4d0*expGau))
>+  y = dsqrt(expGau)*AB
>+
>+  if(AB > 0d0) then 
>+    ss = (1d0+x*(x+y))*derfc(x+y)*dexp(2d0*x*(x+y)) -
>(1d0+x*(x-y))*derfc(x-y)*dexp(2d0*x*(x-y))
>+    ss = ss/y
>+  else
>+    ss =
>(4d0*E**x**2*(1d0+x**2))/dsqrt(Pi)-2d0*E**(2d0*x**2)*x*(3d0+2d0*x**2)*dErfc(x)
>+  endif
>+
>+  k = expSla*dsqrt(expGau)*x**3d0*dexp(-x*x)*4d0*pi/expSla**(3d0/2d0)
>+  ss = k*ss
>+
>+! Print result
>+  write(*,*) ss
>
>Debug output or mandatory?
>
>> +  implicit none
>+  double precision, intent(in)  :: t
>+  double precision              :: pi
>+
>+  pi = 4d0*atan(1d0)
>+
>+  if(t > 0d0) then
>+    BoysF0 = 0.5d0*dsqrt(pi/t)*derf(dsqrt(t))
>+  else
>+    BoysF0 = 1d0
>+  endif
>+
>+end
>+!*****************************************************************************
>+
>+!TODO
>
>Do we really want to merge TODO thing in the main-branch?
>
>> +   endif
>+   is_MPI_master = (MPI_rank == 0)
>+ IRP_ELSE
>+   MPI_rank = 0
>+   MPI_size = 1
>+   is_MPI_master = .True.
>+ IRP_ENDIF
>+
>+
>+END_PROVIDER
>+
>+subroutine qp_mpi_finalize()
>+  implicit none
>+  PROVIDE MPI_Initialized
>+  IRP_IF MPI
>+      integer :: ierr
>
>The Provide MPI_Init doesn't need to go in the IRP_IF to be cleaner?
>Or beter create the function one if IRP_IF MPI (don't know if it's
>possible)
>
>> -  print *, '      ]Zov*XSUXXZXZXSe||vo2       '
>-  print *, '      ]Z#><iiii|i||||==vn2(       '
>-  print *, '      ]Z#i<ii||+|=||=:{no2[       '
>-  print *, '      ]ZUsiiiiivi|=||=vo22[       '
>-  print *, '      ]XZvlliiIi|i=|+|vooo        '
>-  print *, '      =v1llli||||=|||||lii(       '
>-  print *, '      ]iillii||||||||=>=|<        '
>-  print *, '      -ziiiii||||||+||==+>        '
>-  print *, '       -%|+++||=|=+|=|==/         '
>-  print *, '        -a>====+|====-:-          '
>-  print *, '          "~,- --   /-            '
>-  print *, '            -.     )>             '
>-  print *, '           .~      +-             '
>-  print *, '           . .... : .             '
>-  print *, '            -------~              '
>-  print *, ''
>
>Maybe we can add some UTF8 character !
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/LCPQ/quantum_package/pull/201#pullrequestreview-44609695

scemama commented on configure:
```diff
@@ -2,26 +2,18 @@
 # -*- coding: utf-8 -*-
 """configure
 
-Usage: configure <config_file> (--production | --development)
```
Some problems when the installation fails. We had to remove everything and start again from scratch. Development mode is better.

scemama commented on configure:
```diff
@@ -528,7 +517,7 @@ def create_ninja_and_rc(l_installed):
 
-    l = [qp_create_ninja, "create"] + sys.argv[1:]
+    l = [qp_create_ninja, "create", "--development"] + sys.argv[1:]
 
```
yes

scemama commented on ocaml/qp_set_ddci.ml:
```diff
@@ -1,305 +0,0 @@
-open Qputils;;
```
This one is removed from the repository. It is not needed any more.

TApplencourt commented:
Ready to merge? I will trust you for the no-regression test. Travis don't want to work today....

scemama commented:
Fire potato!


On 06/19/2017 03:58 PM, Thomas Applencourt wrote:
>
> Ready to merge? I will trust you for the no-regression test. Travis
> don't want to work today....
>

-- 
Anthony Scemama
IRSAMC / Laboratoire de Chimie et Physique Quantiques
UMR5626 CNRS/UPS
Toulouse

TApplencourt commented:
😲

anbenali commented:
I enjoy these emails..

I will test it intensively today as I will run my big calculations using
the new version

On Mon, Jun 19, 2017 at 10:03 AM, Thomas Applencourt <
notifications@github.com> wrote:

> 😲
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/LCPQ/quantum_package/pull/201#issuecomment-309467921>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AXl6w0ootUjpg15P_FuiTq571kyMHArVks5sFo2-gaJpZM4N8rhc>
> .
>



-- 
-----------------------
Anouar Benali, PhD
Leadership Computing Facility
Argonne National Laboratory
Building 240 Office - 2127
9700 S Cass Av., Argonne Il, 60439
(630) 252-0058
